tests/test_readme.py

```diff
@@ -0,0 +1,21 @@
+import re
+from pathlib import Path
+
+import pytest
+
+
+def _parse_readme_examples():
+    lines = open(Path(__file__).parent.parent / "README.md").read().splitlines()
+
+    # use regex to find all python code blocks
+    code_blocks = re.findall(r"```python(.*?)```", "\n".join(lines), re.DOTALL)
+
+    return code_blocks
+
+
+@pytest.mark.parametrize("codeblock_example", _parse_readme_examples())
+def test_readme_example(codeblock_example: str):
+    """
+    Check that execution of the python code block in the README does not raise an exception.
+    """
+    exec(codeblock_example)
```

Comment:
`exec` is so scary security-wise, but I don't see a problem with it here. We just need to make sure that there's no weird edits to the readme (which we would not allow anyhow)

Reply:
Yes, I agree... I guess I could use the python AST to blacklist/whitelist commands... I will have a quick look :)

Reply:
I went down a rabbit hole with https://github.com/zopefoundation/RestrictedPython... not worth it. I think we should just keep this as is

Reply:
Agreed. I suspect that that rabbit hole goes deep :smile: I think this is perfectly fine as is